nntrainer/app_context.cpp

```diff
@@ -47,8 +51,104 @@
 #include <nnstreamer_layer.h>
 #endif
 
+/// add #ifdef across platform
+static std::string layerlib_suffix = "layer.so";
+static const std::string func_tag = "[AppContext] ";
+
+constexpr const char *DEFAULT_CONF_PATH = "/etc/nntrainer.ini";
+
+constexpr const char *getConfPath() {
+#ifdef NNTRAINER_CONF_PATH
+  return NNTRAINER_CONF_PATH;
+#endif
+  return DEFAULT_CONF_PATH;
+}
+
 namespace nntrainer {
 
+namespace {
+bool endswith(const std::string &target, const std::string &suffix) {
+  if (target.size() < suffix.size()) {
+    return false;
+  }
+  size_t spos = target.size() - suffix.size();
+  return target.substr(spos) == suffix;
+}
+
+/**
+ * @brief Get the plugin path from conf ini
+ *
+ * @return std::string plugin path
+ */
+std::string getPluginPathConf() {
+  std::string conf_path{getConfPath()};
+
+  NNTR_THROW_IF(!isFileExist(conf_path), std::invalid_argument)
+    << func_tag << "There is no existing config file";
+  ml_logd("%s conf path: %s", func_tag.c_str(), conf_path.c_str());
+
+  dictionary *ini = iniparser_load(conf_path.c_str());
+  NNTR_THROW_IF(ini == nullptr, std::runtime_error)
+    << func_tag << "loading ini failed";
+
+  auto freedict = [ini] { iniparser_freedict(ini); };
+
+  const char *path = iniparser_getstring(ini, "plugins:layer", NULL);
```

Comment:
it would be better to have `layer` as an argument.

Reply:
Done.